h2o-py/h2o/model/model_base.py

```diff
@@ -1054,51 +1076,129 @@ def partial_plot(self, data, cols, destination_key=None, nbins=20, weight_column
+                    ax.plot_surface(X, Y, Z, cmap=cm.coolwarm,linewidth=1, antialiased=False, alpha=0.5, edgecolor='k')
                     if plot_stddev:
-                        axs[i, 0].plot(x, lower, 'b--')
-                        axs[i, 0].plot(x, upper, 'b--')
-                    axs[i, 0].set_ylim(min(lower) - 0.1*abs(min(lower)), max(upper) + 0.1*abs(max(upper)))
-                    axs[i, 0].set_xticks(x)
-                    axs[i, 0].set_xticklabels(labels)
-                    axs[i, 0].margins(0.2)
-                else:
-                    x = pp[0]
-                    y = pp[1]
-                    axs[i, 0].plot(x, y, "r-")
-                    if plot_stddev:
-                        axs[i, 0].plot(x, lower, 'b--')
-                        axs[i, 0].plot(x, upper, 'b--')
-                    axs[i, 0].set_xlim(min(x), max(x))
-                    axs[i, 0].set_ylim(min(lower) - 0.1*abs(min(lower)), max(upper) + 0.1*abs(max(upper)))
-
-                axs[i, 0].set_title("Partial Dependence Plot For {}".format(col))
-                axs[i, 0].set_xlabel(pp.col_header[0])
-                axs[i, 0].set_ylabel(pp.col_header[1])
-                axs[i, 0].xaxis.grid()
-                axs[i, 0].yaxis.grid()
-            if len(col) > 1:
+                        ax.plot_surface(X, Y, Zupper, cmap=cm.coolwarm,linewidth=0.2, antialiased=False, alpha=0.3, edgecolor='y')
+                        ax.plot_surface(X, Y, Zlower, cmap=cm.coolwarm,linewidth=0.2, antialiased=False, alpha=0.3, edgecolor='g')
+                    ax.set_xlabel(colPairs[0])
+                    ax.set_xlim(min(x), max(x))
+                    ax.set_ylabel(colPairs[1])
+                    ax.set_ylim(min(y), max(y))
+                    ax.set_zlim(min([min(zupper), min(zlower), min(pp[2])]), max([max(zupper), max(zlower), max(pp[2])]))
+                    ax.set_zlabel('Partial dependence')
+                    titles = '2D partial dependence plot for '+colPairs[0] + ' and '+colPairs[1]
+                    ax.set_title(titles)
+                else:  # plot 1D pdp
+                    col = cols[i]
+                    cat = data[col].isfactor()[0]
+                    upper = [a + b for a, b in zip(pp[1], pp[2]) ]  # pp[1] is mean, pp[2] is std
+                    lower = [a - b for a, b in zip(pp[1], pp[2]) ]
+                    axs = fig.add_subplot(gxs[i])
+                    self.setAxs1D(axs, upper, lower, plot_stddev, cat, pp, 0, col)  # setup graph, axis, labels and ...
+
+            if totFig > 1:
                 fig.tight_layout(pad=0.4, w_pad=0.5, h_pad=1.0)
             if(save_to_file is not None):
                 plt.savefig(save_to_file)
 
         return pps
-
+    
+    # change x, y, z to be 2-D numpy arrays in order to plot it.
+    # note that, x stays at one value for the duration of y value changes.
+    def predFor3D(self, x, y, z, colPairs, nbins, user_cols, user_num_splits):
```

Comment:
Is this a private function? If so, better to append with `__`

Reply:
Yeap, fixed!  Thanks.